src/components/vanilla/charts/CompareLineChart/index.tsx

```diff
@@ -0,0 +1,288 @@
+        mode: 'index',
+        intersect: false
+      },
+      layout: {
+        padding: {
+          left: 0,
+          right: 0,
+          top: props.showLabels ? 20 : 0, // Added so the highest data labels fits
+          bottom: 0
+        }
+      },
+      scales: {
+        y: {
+          min: props.yAxisMin,
+          grace: '0%', // Add percent to add numbers on the y-axis above and below the max and min values
+          grid: {
+            display: false
+          },
+          ticks: {
+            precision: 0
+          },
+          title: {
+            display: !!props.yAxisTitle,
+            text: props.yAxisTitle
+          }
+        },
+        period: {
+          min: bounds.period?.from?.toJSON(),
+          max: bounds.period?.to?.toJSON(),
+          grid: {
+            display: false
+          },
+          type: 'time',
+          time: {
+            round: props.granularity,
+            isoWeekday: true,
+            displayFormats: {
+              month: 'MMM',
+              day: 'd MMM',
+              week: 'd MMM',
+              hour: 'eee HH:mm',
+              minute: 'eee HH:mm',
+              second: 'HH:mm:ss'
+            },
+            unit: props.granularity
+          },
+          title: {
+            display: !!props.xAxisTitle,
+            text: props.xAxisTitle
+          }
+        },
+        comparison: {
+          min: bounds.comparison?.from?.toJSON(),
+          max: bounds.comparison?.to?.toJSON(),
+          display: false,
+          grid: {
+            display: false
+          },
+          type: 'time',
+          time: {
+            round: props.granularity,
+            isoWeekday: true,
+            displayFormats: {
+              month: 'MMM',
+              day: 'd MMM',
+              week: 'd MMM',
+              hour: 'eee HH:mm',
+              minute: 'eee HH:mm',
+              second: 'HH:mm:ss'
+            },
+            unit: props.granularity
+          },
+          title: {
+            display: false
+          }
+        }
+      },
+      animation: {
+        duration: 400,
+        easing: 'linear'
+      },
+      plugins: {
+        legend: {
+          display: props.showLegend,
+          position: 'bottom',
+          labels: {
+            usePointStyle: true,
+            boxHeight: 8
+          }
+        },
+        datalabels: {
+          align: 'top',
+          display: props.showLabels ? 'auto' : false,
+          formatter: (v) => {
+            const val = v ? formatValue(v.y, { type: 'number', dps: props.dps }) : null;
+            return val;
+          }
+        }
+      }
+    };
+  }, [props]);
+
+  useEffect(() => {
+    ChartJS.defaults.plugins.tooltip.callbacks.title = (lines: any[]) => {
+      const definedFormat = {
+        quarter: 'MMM yyyy',
+        month: 'MMM',
+        day: 'd MMM',
+        week: 'd MMM',
+        hour: 'eee HH:mm',
+        minute: 'eee HH:mm',
+        second: 'HH:mm:ss'
+      };
```

Comment:
this is equal to the line 213. lets create a const and reuse it

Reply:
@mad-raccoon great spot. Should have noticed. Done 👍